toolkit/tools/scheduler/schedulerutils/depsolver.go

```diff
@@ -76,8 +100,12 @@ func LeafNodes(pkgGraph *pkggraph.PkgGraph, graphMutex *sync.RWMutex, goalNode *
 		// Ideally we will wait for the actual provider of the implicit node to be built and convert the implicit node to
 		// a normal node via InjectMissingImplicitProvides().
 		if !useCachedImplicit && pkgNode.Implicit && pkgNode.State == pkggraph.StateCached {
-			logger.Log.Debugf("Skipping cached implicit provide leaf node: %v", pkgNode)
-			return
+			if isToolchainPackage(pkgNode, toolchainPackages) {
+				logger.Log.Debugf("Allowing cached implicit toolchain leaf node: %v", pkgNode)
+			} else {
+				logger.Log.Debugf("Skipping cached implicit provide leaf node: %v", pkgNode)
+				return
+			}
```

Comment:
This will probably print a lot of output. What do you think about making it trace level?